src/libcore/array.rs

```diff
@@ -210,6 +214,21 @@ macro_rules! array_impls {
                 }
             }
 
+            #[unstable(feature = "array_into_iter", issue = "0")]
```

Comment:
Trait implementations are still insta-stable, right?

Reply:
Oh, yeah.  I'll need to update the truly unstable items with a tracking issue (if this PR is accepted), so I'll change the impls to stable then.

Reply:
Event if the `IntoIter` type is left as unstable, it will still be reachable through `<<[u8; 0]> as IntoIter>::Iter>`, right?

Reply:
True, you can write something like:

```rust
fn my_iter() -> <[u8; 0] as IntoIterator>::IntoIter {
    [].into_iter()
}
```

I think that's OK.  We could change the associate type entirely without breaking such code.